File: abc238/c.cpp
#include<bits/stdc++.h>
using namespace std;
using ll = long long;

int mod = 998244353;

long long sum(int digit){
  long long f = 9*pow(10,(digit-1));
  ll a = (1+f) % mod;
  f %= mod;
  long long ans = a * f/2;
  ans %= mod;
  return ans;
}

long long sum2(int digit, long long f){
  f = f - pow(10, digit-1)+1;
  ll a = (1+f) % mod;
  f %= mod;
  long long ans = a * f/2;
  ans %= mod;
  return ans;
}


int main() {
  long long n;
  cin >> n;
  int digit=0;
  long long temp = n;
  while(temp!=0){
    temp/=10;
    digit++;
  }
  long long ans = 0;
  for(int i=1; i<digit; i++){
    ans += sum(i);
    ans %= mod;
  }
  ans += sum2(digit, n);
  ans %= mod;
  cout << ans << endl;
  return 0;
}

File: abc238/c.py
n = int(input())
mod = 998244353

def sum1(f):
    f%=mod
    ans = (1+f)*f/2
    ans %= mod
    return ans

def digitf(d):
    ans = pow(10,d) - pow(10,d-1)
    return ans

temp = n
digit = 0
while(temp!=0):
    temp//=10
    digit +=1
ans = 0

for i in range(1, digit):
    f = digitf(i) 
    ans += sum1(f)
    ans %= mod
ff = n-pow(10, digit-1)+1 
ans += sum1(ff) 
ans %= mod 
print(int(ans))
